Add `multiple_closures_with_trailing_closure` rule

erikstrottmann opened the pull request:
As described in #1801, the Multiple Closures with Trailing Closure rule disallows trailing closure syntax when passing more than one closure argument to a function. This is my first PR for SwiftLint, so I’m certainly open to feedback.

Fixes #1801.

marcelofabri commented on Source/SwiftLintFramework/Rules/MultipleClosuresWithTrailingClosureRule.swift:
```diff
@@ -0,0 +1,95 @@
+        return [
+            StyleViolation(ruleDescription: type(of: self).description,
+                           severity: configuration.severity,
+                           location: Location(file: file, byteOffset: call.offset))
```
Do you think it would be better if we added the violation to the trailing closure? For example:

```swift
foo.something(param1: { $0 }) ↓{ $0 + 1 }
```

erikstrottmann replied:
Makes sense to me. I had initially put the violation at the start of the function call because that's where it is for `trailing_closure`, but I think the start of the trailing closure is more precise.

erikstrottmann commented:
Rebased and docs regenerated.

Note that I also added `// swiftlint:disable file_length` to `Tests/SwiftLintFrameworkTests/RulesTests.swift`, because this commit together with the recent changes to master cause that file to exceed the 400 line limit.

marcelofabri commented:
Thanks @erikstrottmann 🚀
